crates/adapters/src/adhoc/table.rs

```diff
@@ -484,47 +498,52 @@ use `with ('materialized' = 'true')` for tables, or `create materialized view` f
                 let mut insert_builder = SendableArrowBuilder::new(sas)?;
 
                 let mut cur_batch_size = 0;
+
                 while cursor.key_valid() {
-                    if !cursor.val_valid() {
-                        cursor.step_key();
-                        continue;
-                    }
-                    let mut w = cursor.weight();
+                    while cursor.val_valid() {
+                        let mut w = cursor.weight();
 
-                    if w < 0 {
-                        cursor.step_key();
-                        panic!("Unexpected key with negative weight encountered while processing ad-hoc query");
-                    }
+                        if w < 0 {
+                            cursor.step_val();
+                            return Err(datafusion::common::DataFusionError::Execution("Unexpected record with negative weight encountered while processing ad-hoc query.".to_string()));
```

Comment:
it looks like a good idea not to kill the pipeline, but it's still a severe bug that we should find about

Reply:
Negative weights can absolutely happen, and DBSP should be ok with them. You might argue it's a user error, but not a bug in the pipeline.